### ISSUE TITLE
research: verify frozen v1 reproduction before merging torch 2.14 (#129)

### ISSUE THREAD

**Issue by hoangnguyen2003** [hoangnguyen2003/edgar-moe#131]
## Problem

Dependabot #129 bumps torch from 2.13.0 to 2.14.0. CI is green, but CI can't tell whether the frozen v1 model still reproduces:

- `scripts/validate_frozen_runtime.py` checks artifact hashes and identities only.
- The locked-test scores and the training dataset are private local data, so CI never re-scores them.

torch runs both the frozen MoE and the FinBERT encoder in the forward runner, so a numerics change would shift prospective scores without failing anything. The FinBERT embedding cache identity also leaves out the torch version, so embeddings cached under 2.13 and computed fresh under 2.14 would mix.

## Baseline (torch 2.13.0, 2026-09-21, macOS arm64)

```sh
uv run python scripts/audit_frozen_normalization.py \
  --selection data/artifacts/walk-forward/research-2026-07-31-3553e7ad78dd/walk-forward-selection.json
```

Result: `status: passed` over 1,794 locked-test events, with `max_saved_score_error: 7.49e-10`.

## Verification

1. Check out #129 and run `uv sync --locked --extra research`.
2. Rerun the audit. It must pass, and `max_saved_score_error` should stay at float32 rounding level.
3. Encode a sample of filings with `FinBertEncoder` under both torch versions. Record the maximum absolute difference in CLS embeddings and sentiment probabilities.
4. Record the numbers here. If they match, merge #129. Otherwise close it and add a Dependabot ignore rule for torch, as #122 did for transformers.

## Scope

The same check applies to any upgrade that touches frozen inference or legacy_v1 features: torch, transformers (majors held by #122), numpy, scikit-learn, and pandas 3.

**Comment by hoangnguyen2003** [comment]
Implemented and squash-merged via PR #164 (merge commit c5de7cd): https://github.com/hoangnguyen2003/edgar-moe/pull/164

The frozen-runtime gate now:
- namespaces FinBERT embedding caches by exact PyTorch and Transformers versions;
- records non-secret runtime versions, reproduced-score digest/shape, and an optional private score archive in audit_frozen_normalization.py;
- compares baseline/candidate reports and score archives offline, binding report metadata to archive digests and failing closed on frozen identity, index, shape, or score drift;
- is covered by unit tests, ADR 0021, README/architecture docs, and CI command checks.

Private current-runtime evidence (not committed): torch 2.13.0, Transformers 4.57.6, 1,794 locked-test events, max saved-score error 9.31e-10; duplicate-control comparison passed with max absolute score delta 0.

Issue remains open: this proves the gate and current baseline only. The Dependabot torch 2.14 candidate still needs its own environment reproduction and comparison before PR #129 can be accepted; a passing comparison is necessary evidence, not permission to change frozen v1.

**Comment by hoangnguyen2003** [comment]
Completed the actual candidate reproduction after PR #164.

Using a detached worktree at the PR #129 torch 2.14.0 lockfile with an isolated uv environment:
- baseline runtime: torch 2.13.0, Transformers 4.57.6, NumPy 2.5.1, scikit-learn 1.9.0, pandas 2.3.3
- candidate runtime: torch 2.14.0, Transformers 4.57.6, NumPy 2.5.1, scikit-learn 1.9.0, pandas 2.3.3
- frozen artifact SHA, champion parameters, target normalization, train/test counts (4,003/1,794), and test indices matched
- both audits passed with max saved-score error 9.31e-10
- offline comparator passed with max absolute score delta 0.0; score digest matched

This is private evidence; no raw data or score archive was committed. The torch 2.14 candidate is numerically compatible with frozen v1 under the configured tolerance. I will now verify PR #129's normal CI/status checks before squash-merging it. Issue can be closed after that merge and post-merge checks.

**Comment by hoangnguyen2003** [comment]
Resolved by PR #129 (squash merge 4553828) after the private torch 2.13→2.14 frozen-v1 reproduction passed with max absolute score delta 0.0. Post-merge CI, CodeQL, deployment smoke, dependency-graph update, and Vercel all passed. Compatibility-gate implementation and evidence trail are in PR #164 and the comments above.
